library/src/scala/quoted/Toolbox.scala

```diff
@@ -16,12 +16,35 @@ object Toolbox {
     implicit def make(implicit settings: Settings): Toolbox = Toolbox.make
   }
 
-  def make(implicit settings: Settings): Toolbox = {
-    val cl = getClass.getClassLoader
+  /** Create a new instance of the toolbox.
+    *
+    * This instance of the toolbox tries to recover the classloader of the application based on
+    * the classloader of the class that call make. This may not always be the correct classloader,
+    * in which case the classloader must be passed explicitly.
+    *
+    * @param settings toolbox settings
+    * @return A new instance of the toolbox
+    */
+  @forceInline def make(implicit settings: Settings): Toolbox = {
```

Comment:
@smarter could you have a look at this method?

Reply:
Looks super hacky to me. I'd like to know what scalac does.

Reply:
They bailed out on trying to recover it. They basically have the equivalent of the other `make` that asks for the class loader explicitly. In the docs they just tell the user to pass `getClass.classLoader`

Reply:
I will remove this variant for now.

Reply:
This hacky method was removed